### DISCUSSION TITLE
Using this with Griffe

### DISCUSSION THREAD

**Discussion by patrick91** [mkdocstrings/griffe-typingdoc#6] [Q&A] [answered]
Hi there 👋

Would it be possible to use this with https://github.com/mkdocstrings/griffe? My documentation site doesn't use Mkdocs at the moment so I'd love to build the API docs on top of griffe and try the `Annotated+Doc` approach!

**Comment by pawamoy** [comment, marked as answer]
Hi!

Of course! We document the use of extensions here: https://mkdocstrings.github.io/griffe/extensions/. I don't know what tool you use to build your docs, so I'm not sure it supports passing extensions. I can chime in issues you open there :slightly_smiling_face:

**Reply by patrick91** [reply]
uh, I totally missed that! I'll try and report back 😊

**Reply by patrick91** [reply]
When I run `griffe` with the extension like this I don't get the doc added anywhere (except for the type annotation):

```bash
griffe dump -e griffe_typingdoc strawberry
```

See example here:

<img width="824" alt="CleanShot 2023-10-25 at 04 23 22@2x" src="https://github.com/mkdocstrings/griffe-typingdoc/assets/667029/887a3cdd-23bd-43e4-af99-6168a23fa696">

should it do that, or did I misunderstand how it works?

**Reply by pawamoy** [reply]
The extension actually inserts the `typing.Doc` value into the *parsed* docstring, not the original one (as to not lie about the source). To dump parsed docstrings, you need the `-f` (full) flag. You can optionally specify the docstring style with the `-d` flag (like `-d google`). If the tool you use to build docs is written in Python, note that it can access the parsed docstring too, while rendering docs: https://mkdocstrings.github.io/griffe/reference/griffe/dataclasses/#griffe.dataclasses.Docstring.parsed.

This is especially true for parameters, which originally (given Python's features) cannot have a docstring of their own.

But now that you mention it, maybe we could consider the `typing.Doc` value to *be* the actual, original docstring **for attributes**.

**Reply by patrick91** [reply]
mmh, it seems to be same with `-f`: 

<img width="811" alt="CleanShot 2023-10-25 at 14 49 27@2x" src="https://github.com/mkdocstrings/griffe-typingdoc/assets/667029/94521838-af1b-4cc3-b650-9aa02b32245e">

**Reply by pawamoy** [reply]
Have you pushed your changes to a branch so that I can try?

**Reply by pawamoy** [reply]
Or just post the contents of your screenshot above, I'll paste it locally :slightly_smiling_face:

**Reply by patrick91** [reply]
just pushed to `feature/docs`!

**Reply by pawamoy** [reply]
Thanks!

There's a bug in Griffe :smile: Or at least in the interop between full dumps and extensions adding sections to parsed docstrings. Working on a fix :slightly_smiling_face:

**Reply by pawamoy** [reply]
Fixed in 0.36.8.

**Reply by patrick91** [reply]
awesome! thanks, that works now 😊 I'll let you know when I have implemented this in Strawberry's new website :D
